Src/main.c

```diff
@@ -70,6 +70,11 @@ void SystemClock_Config(void);
 int main(void)
 {
 
+  int adcValue;
+  double adcPercentage;
+  double pwmValue;
+  int HIGH = 2;
+  int LOW = 1;
```

Comment:
`HIGH` and `LOW` should be `const`, right? we don't want to overwrite them accidentally in the future! It would probably be worth it to have a more descriptive name as well, so that the math that you do below is easier to follow on first read.

Also review the data types you use. `int` is perhaps larger than required by `HAL_ADC_GetValue()`.